File: zmin9/BOJ_10610번_30_Greedy.js
// 입력 처리
const fs = require('fs');

// 아무리 봐도 알고리즘이 틀린게 없는데 틀렸대서 봤더니 입력 부분의 문제였던 것 같다..
// 주석 처리된 것처럼 해도 값도 잘 나오고 알고리즘에 들어가게 되는 변수의 구조도 동일해보이는데 왜 안됐던건지 잘 모르겠음
//let N = fs.readFileSync('./dev/stdin').toString().split('');
let [input] = fs.readFileSync('./input.txt').toString().trim().split('\n');
const N = input.split('');



// 30의 배수이려면 1의 자릿수가 0이어야하고 모든 자릿수를 더한 값이 3의 배수
if(!N.includes('0')) {
    console.log(-1);
}
else {
    let sum = 0;
    N.forEach(n=>{
        sum += parseInt(n) % 3;
    });
    if (sum % 3 !== 0){
        console.log('-1');
    }
    else {
        console.log(N.sort().reverse().join(''));
    }
}

File: jieun^,^/BOJ_10610번_30_Greedy.py
# N의 순서를 바꿔서 30의 배수가 되는 수 중 가장 큰 수를 만들기
# 30의 배수가 아예 존재하지 않는다면 -1

# 간단하게 생각하면..? 만들 수 있는 모든 수, 가장 큰 수 -> 작은 수, 30배수인지 확인

# 하.지.만.
# 10의 배수 -> 1의 자리는 무.조.건. 0
# 3의 배수 -> 각 자리 수의 합이 무.조.건. 3의 배수
# => N에 0이 포함되어 있고, 자릿수의 합이 3의 배수라면 무.조.건. 가능

# => 가장 큰 수 = 높은 순으로 sort한 수

def solution():
    N = input()    # 마르코가 만난 양수 N
    ans = 0 # 정답 0으로 초기화

    if '0' in N:    # 10배수
        arr = []
        for n in N:
            arr.append(int(n))

        if sum(arr)%3==0:   # 3배수
            arr.sort(reverse=True)
            ans_str = ''
            for a in arr:
                ans_str += str(a)
            ans = int(ans_str)
 
        else: ans = -1  # 3배수 X
        
    else:       # 10배수 X
        ans = -1

    print(ans)

solution()

# 시간 초과 나오길래.. sort가 오래 걸리나? 해서 퀵 sort 까지 해봤는데 .. 여전히 시간초과
# 아마도 N을 배열에 넣을 때, %10으로 각 자리 수를 넣었는데, 그게 오래 걸렸던 것 같다